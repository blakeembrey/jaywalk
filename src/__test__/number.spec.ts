import { validationFunction } from '../'

describe('Number', () => {
  it('should validate numbers', () => {
    const validate = validationFunction({
      '@type': 'Number'
    })

    const input = 101

    return validate(input).then(function (output) {
      expect(output).toEqual(input)
    })
  })

  it('should reject non-numbers', () => {
    const validate = validationFunction({
      '@type': 'Number'
    })

    expect.assertions(1)

    return validate(true).catch(function (err) {
      expect(err.errors).toMatchSnapshot()
    })
  })
})
