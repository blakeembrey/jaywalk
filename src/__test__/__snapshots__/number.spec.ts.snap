exports[`Number should reject non-numbers 1`] = `
Array [
  Object {
    "dataPath": Array [],
    "message": "not a number",
    "schemaPath": Array [
      "@type",
    ],
  },
]
`;
